### DISCUSSION TITLE
how are BLOCKS made?

### DISCUSSION THREAD

**Discussion by TanjIsGray** [The-OpenROAD-Project/OpenROAD#4250] [General]
I have a design which uses multiples of 4 blocks or macros which were designed to be individually optimized and then placed by abutment.  I can get pretty decent runs on one block at a time but when I try to use the blocks in the whole design all that comes out is the first listed block.

Looking through examples I see similar behavior with AES.  The whole design does evidence hierarchy with fuzzy clouds for the repeated blocks, but they are not macros using the block design.  If aes-block is run it generates a nice compact aes_rcon but ignores aes_sbox, and that is all it generates.  Not a whole AES pipe.

I tried the mock-array example but the results look nothing like I see in discussions here.  Probably something I have not got right in chisel or other parts.  Anyway the build is so indirect that I could spend a long time puzzling over it and never fix it or learn about blocks.

It looks to me like AES is intended to work the same kind of design, with two tightly optimized blocks that are then repeated.  Are other people seeing it do that?  Any hints why I see no blocks incorporated in the aes project, and only one lone block in aes-block with everything else missing?

**Comment by maliberty** [comment]
Is this what you see for aes-blocks?
![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/761514/56fe74b4-26ef-4603-9fb6-4c53c158bf6c)
